src/uct/ib/base/ib_device.c

```diff
@@ -378,6 +378,51 @@ ucs_status_t uct_ib_device_query(uct_ib_device_t *dev,
     return UCS_OK;
 }
 
+static ucs_status_t uct_ib_device_gid_hash_init(uct_ib_device_t *dev)
+{
+    static const union ibv_gid zero_gid = { .raw = {0} };
+    uct_ib_device_gid_info_t gid_info;
+    int port, gid_idx, gid_tbl_len, kh_ret;
+    ucs_status_t status;
+    char gid_str[128];
+
+    kh_init_inplace(uct_ib_gid, &dev->gid_hash);
+
+    for (port = dev->first_port; port < (dev->first_port + dev->num_ports);
+         ++port) {
+        gid_tbl_len = uct_ib_device_port_attr(dev, port)->gid_tbl_len;
+        for (gid_idx = 0; gid_idx < gid_tbl_len; ++gid_idx) {
+            status = uct_ib_device_query_gid_info(dev->ibv_context,
+                                                  uct_ib_device_name(dev),
+                                                  port, gid_idx, &gid_info);
+            if (status != UCS_OK) {
+                goto err;
+            }
+
+            if (!memcmp(&gid_info.gid, &zero_gid, sizeof(zero_gid))) {
```

Comment:
there should not be padding errors, right?

Reply:
rigth, `ibv_query_gid` fill unused GID's bytes with `0`